main/main.go

```diff
@@ -270,153 +380,239 @@ func generateDataFiles() error {
 	doneChan := make(chan bool, 1)
 	go utils.WaitForWaitGroup(waitGroup, doneChan)
 
 	select {
 	case err := <-errChan:
-		fmt.Printf("Stop diff generation due to error from dcp client %v\n", err)
+		difftool.logger.Errorf("Stop diff generation due to error from dcp client %v\n", err)
 		err1 := sourceDcpDriver.Stop()
 		if err1 != nil {
-			fmt.Printf("Error stopping source dcp client. err=%v\n", err1)
+			difftool.logger.Errorf("Error stopping source dcp client. err=%v\n", err1)
 		}
 		err1 = targetDcpDriver.Stop()
 		if err1 != nil {
-			fmt.Printf("Error stopping target dcp client. err=%v\n", err1)
+			difftool.logger.Errorf("Error stopping target dcp client. err=%v\n", err1)
 		}
 		return err
 	case <-doneChan:
-		fmt.Printf("Source cluster and target cluster have completed\n")
+		difftool.logger.Infof("Source cluster and target cluster have completed\n")
 		return nil
 	}
 
 	return nil
 }
 
-func waitForDuration(sourceDcpDriver, targetDcpDriver *dcp.DcpDriver, errChan chan error, duration uint64, delayDurationBetweenSourceAndTarget time.Duration) (err error) {
+func (difftool *xdcrDiffTool) waitForDuration(sourceDcpDriver, targetDcpDriver *dcp.DcpDriver, errChan chan error, duration uint64, delayDurationBetweenSourceAndTarget time.Duration) (err error) {
 	timer := time.NewTimer(time.Duration(duration) * time.Second)
 
 	select {
 	case err = <-errChan:
-		fmt.Printf("Stop diff generation due to error from dcp client %v\n", err)
+		difftool.logger.Errorf("Stop diff generation due to error from dcp client %v\n", err)
 	case <-timer.C:
-		fmt.Printf("Stop diff generation after specified processing duration\n")
+		difftool.logger.Infof("Stop diff generation after specified processing duration\n")
 	}
 
 	err1 := sourceDcpDriver.Stop()
 	if err1 != nil {
-		fmt.Printf("Error stopping source dcp client. err=%v\n", err1)
+		difftool.logger.Errorf("Error stopping source dcp client. err=%v\n", err1)
 	}
 
 	time.Sleep(delayDurationBetweenSourceAndTarget)
 
 	err1 = targetDcpDriver.Stop()
 	if err1 != nil {
-		fmt.Printf("Error stopping target dcp client. err=%v\n", err1)
+		difftool.logger.Errorf("Error stopping target dcp client. err=%v\n", err1)
 	}
 
 	return err
 }
+
+func (difftool *xdcrDiffTool) retrieveReplicationSpecInfo() error {
+	// CBAUTH has already been setup
+	rcMap, err := difftool.remoteClusterSvc.RemoteClusters()
+	if err != nil {
+		difftool.logger.Errorf("Error retrieving remote clusters: %v\n", err)
+		return err
+	}
+
+	specMap, err := difftool.replicationSpecSvc.AllReplicationSpecs()
+	if err != nil {
+		difftool.logger.Errorf("Error retrieving specs: %v\n", err)
+	}
+
+	for _, ref := range rcMap {
+		if ref.Name_ == options.remoteClusterName {
+			difftool.specifiedRef = ref
+			break
+		}
+	}
+
+	for _, spec := range specMap {
+		if spec.SourceBucketName == options.sourceBucketName && spec.TargetBucketName == options.targetBucketName {
```

Comment:
not safe. need to compare UUIDs

Reply:
Done. Will check spec's targetClusterUUID with ref's UUID